main.go

```diff
@@ -50,6 +52,7 @@ type Config struct {
 	FilesToRedact                 string `env:"files_to_redact"`
 	DebugMode                     bool   `env:"debug_mode,opt[true,false]"`
 	BundletoolVersion             string `env:"bundletool_version,required"`
+	UploadConcurrency             string `env:"BITRISE_DEPLOY_UPLOAD_CONCURRENCY"`
```

Comment:
What do you think about generalizing this env var for all steps? Now that I'm facing a very similar problem in the Appcenter upload code, I think it would make sense to have a single env var that controls all step upload concurrencies in builds. This would give a tool to individual users experiencing network flakiness.

Reply:
We use this as an experiment to determine the right value for the step without killing our network bandwidth. If other step would use this then we could not get accurate results from an experiment because we would control multiple variable part at the same time. Also different steps might have different upload characteristics so they might need different values. 

On the other hand I could see a potential in such a value. And actually as I was writing this message I think I iterated on this feature and have an improved idea. 😄

Based on everything I wrote in the first section I think what would be great is if we could easily add new build level env vars without modifying the backend. Like an option in Optimus Prime where we can add new env vars and also we could specify which experiment is backing that env var. That way we could easily tweak for example the concurrency of every step individually.